dlt/sources/helpers/rest_client/client.py

```diff
@@ -58,6 +58,7 @@ class RESTClient:
         auth (Optional[AuthBase]): Authentication configuration for all requests.
         paginator (Optional[BasePaginator]): Default paginator for handling paginated responses.
         data_selector (Optional[jsonpath.TJsonPath]): JSONPath selector for extracting data from responses.
+            Only used when paginating. Use `extract_response` for other requests.
```

Comment:
Let's remove the suggestion to use `extract_response` as it could be actually simpler to directly get the data.
```suggestion
            Only used in `paginate`.
```

Reply:
> Let's remove the suggestion to use extract_response as it could be actually simpler to directly get the data.

There's a lot of other lifting in that function though, I think it's actually quite good to use it, but I don't mind either way, committed.